<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <node name="my_publisher" pkg="tedusar_detect_evaluation" type="image_publisher" output="screen">

    <!-- folder, pictures -->
    <param name="dir" type="string" value="$(env HOME)/Dropbox/evaluation/pictures/testing_pic/" />
    <param name="subscribe" type="string" value="/features" />
    <param name="pub_rect" type="string" value="/tooverlapping" />
    <param name="pub_roc" type="string" value="/toroc" />
    <param name="pub_cnn" type="string" value="/tocnn" />

    <!-- For additional debug output -->
    <param name="debug" type="bool" value="true"/>
    <param name="cnn" type="bool" value="true"/>

  </node>
  
  <node name="tedusar_cnn_prediction_node" pkg="tedusar_cnn_detection" type="tedusar_cnn_prediction_node.py" output="screen">

    <!-- Overfeat network -->
    <param name="overfeat_weights" type="string" value="$(env HOME)/data/overfeat/net_weight_0"/>

    <!-- Camera -->
    <param name="subscribe_cam" type="string" value="/tocnn" />

    <!-- Load pretrained classifier -->
    <param name="trained_directory" type="string" value="$(find tedusar_cnn_detection)/trained/"/>
    <param name="load_name" type="string" value="cnn_trained_victims"/>

    <!-- For additional debug output -->
    <param name="debug" type="bool" value="true"/>

    <!-- Use for local, no ray cast only testing -->
    <param name="local" type="bool" value="true"/>
    <param name="subscribe_local_cam" type="string" value="/tocnn"/>

  </node>


  <node name="roc_choice" pkg="tedusar_detect_evaluation" type="roc_choice" output="screen">
  
    <!-- folder, pictures -->
    <param name="plot" type="string" value="$(env HOME)/Dropbox/evaluation/pythonScripts/roc_curve.py $(env HOME)/Dropbox/evaluation/pythonScripts/score" />
    <param name="xml_files" type="string" value="$(env HOME)/Dropbox/evaluation/xml/testing_xml/" />
    <param name="score" type="string" value="$(env HOME)/Dropbox/evaluation/pythonScripts/score" />
    <param name="subscribe" type="string" value="/toroc" />
    <param name="subscribecnn" type="string" value="/toroc" />

    <!-- For additional debug output -->
    <param name="debug" type="bool" value="false"/>
    <param name="cnn" type="bool" value="true"/>

  </node>

</launch>
